### ISSUE TITLE
Wheels to install in a Python 3.12 envionment

### ISSUE THREAD

**Issue by RNarayan73** [y-crdt/ypy#152]
Following error occurs when trying to install in a Python 3.12 environment:

> (jupy312) PS C:\Users\naray\Python\venvs\jupy312> pip install y-py
> 
> Collecting y-py
>   Downloading y_py-0.6.2.tar.gz (53 kB)
>      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.0/53.0 kB 248.7 kB/s eta 0:00:00
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... done
>   Preparing metadata (pyproject.toml) ... error
>   error: subprocess-exited-with-error
> 
>   × Preparing metadata (pyproject.toml) did not run successfully.
>   │ exit code: 1
>   ╰─> [6 lines of output]
> 
>       Cargo, the Rust package manager, is not installed or is not on PATH.
>       This package requires Rust and Cargo to compile extensions. Install it through
>       the system's package manager or via https://rustup.rs/
> 
>       Checking for Rust toolchain....
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: metadata-generation-failed
> 
> × Encountered error while generating package metadata.
> ╰─> See above for output.
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for details.

Looks like it needs new wheels for Python 3.12 as this isn't an issue with 3.11

**Comment by RNarayan73** [comment]
@dmonad, looks like #138 already made updates for Python 3.12 back in Sept '23.
Is there an eta to release this with Py 3.12 wheels?

**Comment by frenzymadness** [comment]
This package is no longer available in Fedora Linux but when it was, we were building version 0.6.0 with Python 3.12 without any issues. I'm sorry but I have no info about Windows.

**Comment by davidbrochart** [comment]
@RNarayan73 I'm not maintaining this package anymore, since I started [pycrdt](https://github.com/jupyter-server/pycrdt). If you can afford an API change, it might be a better solution for you.

**Comment by RNarayan73** [comment]
> @RNarayan73 I'm not maintaining this package anymore, since I started [pycrdt](https://github.com/jupyter-server/pycrdt). If you can afford an API change, it might be a better solution for you.

As mentioned in https://github.com/jupyterlab/jupyterlab/issues/15332 the issues seems to be a dependency that jupyterlab 3 has on jupyter-server-ydoc.

**Comment by leonardbiofi** [comment]
> [@RNarayan73](https://github.com/RNarayan73) I'm not maintaining this package anymore, since I started [pycrdt](https://github.com/jupyter-server/pycrdt). If you can afford an API change, it might be a better solution for you.


@davidbrochart  

Does that mean that ypy-websocket is also unmaintained at the moment ? Do you have any plan to migrate ypy-websocket to pycrdt ? 

I am currently using ypy-websocket with the django channels consumer. Is there any plan in the future to implement these features in pycrdt directly ? 

Thank you very much

**Comment by davidbrochart** [comment]
Yes, `ypy-websocket` is unmaintained. You can use [pycrdt-websocket](https://github.com/jupyter-server/pycrdt-websocket) instead. It has the Django channels consumer too.
